### PR TITLE
Add Angular guide

### PR DESCRIPTION
**Pull request by simondel**
This guide is meant to help users who want to use Stryker with an Angular project.

### CODE REVIEW

**Comment by simondel** [comment]
@nicojs @Archcry @blabno feedback is very welcome!

**Comment by blabno** [comment]
For me the default dependencies generated by `ng new` were too old and stryker told me only about one of them:
```
 "@angular-devkit/build-angular": "~0.6.8",
 "@angular/cli": "~6.0.8",
```
I had to update both to:
```
"@angular-devkit/build-angular": "~0.7.2",
"@angular/cli": "~6.1.0",
```

Configuring stryker is also not intuitive. I thought that since my tests use jasmine then I should set `testFramework: "jasmine"`, but that leads to
```
SyntaxError: 'import' and 'export' may appear only with 'sourceType: "module"' (1:0)
```

Next thing was `karma.project` property in `stryker.conf.js`. The documentation on home page https://github.com/stryker-mutator/stryker/tree/master/packages/stryker#readme did not mention how that `karma` part shold be configured. I know you can reach it by going to https://stryker-mutator.io/plugins.html and then to `stryker-karma-runner` readme on github, but it's not intuitive.
My intuition told me to set it to the name of my project and I didn't get message that the value is invalid. Instead I got cryptic message:
> Error: The '@angular-devkit/build-angular/plugins/karma' karma plugin is meant to be used from within Angular CLI and will not work correctly outside of it
which was very misleading as my next attempt was trying to remove that plugin from `karma.conf.js`.

File: stryker/angular/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Angular
+Stryker supports Angular projects using the Angular CLI starting from @angular/cli v6.1.0. Are you using an older version? Then there are some tips later in this document.
+
+## @angular/cli 6.1.0 and higher
```
Comment (blabno) [inline stryker/angular/README.md]: You have to mention minimum valid version of `@angular-devkit/build-angular` cause `~0.6.8` was causing stryker to hang in the middle of tests.

File: stryker/angular/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+### Configuration
+After installing the recommended packages, create the `stryker.conf.js` file in your repository. 
+The configuration below contains a good starting point for Angular projects. 
+You may have to change some paths or config settings like the selected browsers. 
+We highly suggest using a headless browser when testing using stryker.
+
+Coverage analysis with [stryker-jasmine](http://npmjs.com/package/stryker-jasmine) is unfortunately not supported as of right now.
+```js
+module.exports = function (config) {
```
Comment (blabno) [inline stryker/angular/README.md]: This sample config will save people lots of time, good job!

**Comment by nicojs** [comment]
Great idea to document this further, but shouldn't this be placed on our wiki? See https://github.com/stryker-mutator/stryker/wiki/Configuration . This is a start for working with presets: https://github.com/stryker-mutator/stryker/issues/1049

So should be moved to the Stryker-handbook as discussed in that issue. Right @_simondel? Just want to make sure we keep things DRY, I don't want to end up with multiple places

**Comment by simondel** [comment]
@nicojs as we recently discussed, the wiki cannot be properly used to collaborate with others due to lacking PR support. 

I'd like to see the documentation in the wiki be migrated to this repo. Then we can point to this repo from the wiki (just like TypeScript does).

**Comment by nicojs** [comment]
That's what I mean. Can we move this PR to the new Stryker-Handbook?

**Comment by simondel** [comment]
@nicojs this PR is on the new stryker-handbook repo 🙈

**Comment by nicojs** [comment]
Wow cool! Didn't realize. I'm on vacation and only have my phone. Proceed.

File: stryker/angular/README.md (hunk 2 above)
Comment (nicojs) [reply stryker/angular/README.md]: How can we scrape this sample during `stryker init`?

**Comment by nicojs** [comment]
> You have to mention minimum valid version of @angular-devkit/build-angular cause ~0.6.8 was causing stryker to hang in the middle of tests.

@blabno I didn't realize that the build-angular dependency was not automatically updated with @angular/cli. Should we also add a runtime check like we have for the cli?

File: stryker/angular/README.md (hunk 1 above)
Comment (simondel) [reply stryker/angular/README.md]: I have documented the dependencies 👍

File: stryker/angular/README.md (hunk 2 above)
Comment (simondel) [reply stryker/angular/README.md]: As of right now, we can't. We can always refactor these documents later so the code could also be used during stryker init.

File: stryker/angular/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      'src/**/*.ts',
+      '!src/**/*.spec.ts',
+      '!src/test.ts',
+      '!src/environments/*.ts'
+    ],
+    mutator: 'typescript',
+    testRunner: 'karma',
+    karma: {
+      configFile: 'src/karma.conf.js',
+      project: 'angular-cli',
```
Comment (nrg500) [inline stryker/angular/README.md]: To avoid confusion about the intended purpose of this variable, this could be renamed to projectType .

Comment (simondel) [reply stryker/angular/README.md]: Good point! This could also prevent the confusion @blabno had.

**Comment by blabno** [comment]
@nicojs when I've read info about upgrading angular cli i've just bumped the version inside package.json, so if there is another way that bumps both cli and devkit then probably it should be documented and printed with that error message.

**Comment by nicojs** [comment]
I'm happy, let's do this

**Pull request merged**